Publish sql microsoft.sql 2025 02 01 preview

mitesh-pv opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.
- [x] A [release plan](https://aka.ms/azsdkdocs/release-plans) has been created. If not, please create one as it will help guide you through the REST API and SDK creation process. 

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.
- For guidance on SDK breaking change review, refer to https://aka.ms/ci-fix.

mitesh-pv committed:
Create base of release-sql-Microsoft.Sql-2025-02-01-preview via Copy-ApiVersion.ps1 (#36370)

* Copy files from preview/2024-11-01-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to preview/2025-02-01-preview

Updated the API version from preview/2024-11-01-preview to preview/2025-02-01-preview.

* Added tag for 2025-02-01-preview in readme file

---------

Co-authored-by: Mitesh Vishwakarma <mvishwakarma@microsoft.com>

skpadhy33 committed:
Swagger changes for Paginated List DDM rules API (#36819)

Co-authored-by: Santosh Padhy <santoshpadhy@microsoft.com>

hareeshghk committed:
Datasync UAMI support API in 2025-02-01-preview (#35876)

* Datasync UAMI support API in 2025-02-01-preview

* Updated json to fix validation failure

* Spell, lint fix

* Fix error message

* Updating error codes

---------

Co-authored-by: Hareesh Kumar Gajulapalli <hgajulapalli@microsoft.com>

mitesh-pv committed:
Azure SQL API Parity (#38442)

* Azure SQL API Parity

* avocado error update

* fixes lint diff errors

---------

Co-authored-by: mvishwakarma <mvishwakarma@microsoft.com>

hareeshghk committed:
Removing datasync breaking change because of required fields (#39059)

Co-authored-by: Hareesh Kumar Gajulapalli <hgajulapalli@microsoft.com>

mitesh-pv committed:
Merge branch 'main' into publish-sql-Microsoft.Sql-2025-02-01-preview

skpadhy33 committed:
Added back the legacy params (#39317)

Co-authored-by: Santosh Padhy <santoshpadhy@microsoft.com>

mitesh-pv committed:
Merge branch 'main' into publish-sql-Microsoft.Sql-2025-02-01-preview

josefree commented:
All active Azure services must convert to TypeSpec by March 31, 2026. Starting July 1, 2026, SDK generation from the autorest toolchain will be blocked. Please reach out [tspconversion@service.microsoft.com](mailto:tspconversion@service.microsoft.com) for TypeSpec conversion support.

mitesh-pv committed:
20250201-preview breaking change fix for databases.json (#40027)

* breaking change fix

* fix model validation failure

---------

Co-authored-by: mvishwakarma <mvishwakarma@microsoft.com>

mitesh-pv committed:
fixing model validation errors (#40364)

* fixing model validation errors

* adds model validation fix

* fixes model validation errors

* Fixes model validation errors

* fixes model validation failures

* model valdation fixes '

* model validation fix

* model validation fixes

* fixes model validation errors

---------

Co-authored-by: mvishwakarma <mvishwakarma@microsoft.com>

zedy-wj committed:
resolving conflict from folder migration

zedy-wj committed:
fix semantic error

zedy-wj committed:
fix semantic error

zedy-wj committed:
add v2

mitesh-pv committed:
fix breaking change (#40593)

Co-authored-by: mvishwakarma <mvishwakarma@microsoft.com>

mikekistler commented:
Breaking changes previously reviewed and approved in #35794

jeremyfrosti committed:
Adding suppressions previously approved in #38470 (#40764)

jeremyfrosti committed:
Removing the list import export operation results from Servers API (#40836)

* Adding suppressions previously approved in #38470

* Removing the list import export operation results from Servers API

mitesh-pv committed:
Fix for lint-diff errors (#40792)

* Fix for lint-diff errors

* Rectifies suppression for ResourceNamePattern

* Remove ResourceNameRestriction validation code

Removed ResourceNameRestriction code from parameters.

* Fix formatting issue

Fix formatting issue

---------

Co-authored-by: mvishwakarma <mvishwakarma@microsoft.com>
Co-authored-by: Jeremy Frosti <111934615+jeremyfrosti@users.noreply.github.com>

jeremyfrosti committed:
Add suppression for ElasticPools to match existing suppression in Databases (#40862)

* Adding suppressions previously approved in #38470

* Removing the list import export operation results from Servers API

* Adding suppression for new currentSku in ElasticPool update

jeremyfrosti commented:
For the LintDiff issues, we addressed them through multiple steps (that will need approval from ARM as well)

First, this very large commit took the suppressions that were added for all the existing LintDiffs in the SQL APIs. 
https://github.com/Azure/azure-rest-api-specs/pull/38646/changes/66a329b0dcb07bfd85d4fe01c9b045d24f52fc99

These suppressions were generated based on the 2024-11-01-preview version as part of the ongoing 2025-01-01 stable version release. The suppressions were approved by ARM in a PR targeting our release branch, but they haven't quite hit main yet, so still need approval here as well. The commit with original approval is here: https://github.com/Azure/azure-rest-api-specs/pull/38470

That change took us down to just 4 LintDiff errors that were caused by this version.

Two of the lintdiffs were caused by the inclusion of a new API under Servers.json. After review, I discovered there were some flaws in the design, and it will need a bit of reworking, and given our timeline I have decided to remove that API and push it to a later version, mitigating those two.

One of the lintdiffs was caused by changes to an enum, upon investigation, this appeared to be a manual error from someone not properly copying over an generation-only attribute to the new version. This was corrected on our side and in this commit: https://github.com/Azure/azure-rest-api-specs/pull/38646/changes/5e1ca835165699ad9b29ddc28c9c0112d569685b

The final lintdiff was caused by the new 'currentSku' property in the ElasticPool resource. This new property was pushed by our customers and PMs to bring ElasticPools in line with Databases, as Databases already has this same property setup and in use for some time. I've added a suppression for this here: https://github.com/Azure/azure-rest-api-specs/pull/38646/changes/a3a1197f598f91057e4c6241078138ccfc123b06

Which matches the other suppression for databases found on line 5104: https://github.com/Azure/azure-rest-api-specs/blob/a3a1197f598f91057e4c6241078138ccfc123b06/specification/sql/resource-manager/Microsoft.Sql/SQL/readme.md?plain=1#L5104

There are no remaining LintDiff errors, but we are now asking for SuppressionApproval on the above basis, would it be possible to have that granted?

jeremyfrosti commented on specification/sql/resource-manager/Microsoft.Sql/SQL/preview/2025-02-01-preview/Jobs.json:
```diff
@@ -0,0 +1,372 @@
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  }
```
This did not change with this new version. https://github.com/Azure/azure-rest-api-specs/blob/e229d6b4b621f5a4f173b6711bc553bbf8556c5a/specification/sql/resource-manager/Microsoft.Sql/SQL/preview/2024-11-01-preview/Jobs.json#L361

jeremyfrosti commented on specification/sql/resource-manager/Microsoft.Sql/SQL/preview/2025-02-01-preview/Jobs.json:
```diff
@@ -0,0 +1,372 @@
+        "endTime": {
+          "format": "date-time",
+          "description": "Schedule end time.",
+          "default": "9999-12-31T17:29:59+05:30",
```
This did not change in this version https://github.com/Azure/azure-rest-api-specs/blob/e229d6b4b621f5a4f173b6711bc553bbf8556c5a/specification/sql/resource-manager/Microsoft.Sql/SQL/preview/2024-11-01-preview/Jobs.json#L324

jeremyfrosti commented on specification/sql/resource-manager/Microsoft.Sql/SQL/preview/2025-02-01-preview/Jobs.json:
```diff
@@ -0,0 +1,372 @@
+          {
+            "name": "jobName",
+            "in": "path",
+            "description": "The name of the job to get.",
```
This did not change in the new version https://github.com/Azure/azure-rest-api-specs/blob/e229d6b4b621f5a4f173b6711bc553bbf8556c5a/specification/sql/resource-manager/Microsoft.Sql/SQL/preview/2024-11-01-preview/Jobs.json#L149

jeremyfrosti commented:
> Please respond/address the copilot comments.

@psah434  I have responded to all of the copilot comments. None of the comments were pertaining to changes in this version, they were all about existing things in our API. We can't go back and modify this all now for this version unfortunately. We will be following the proper "release" branch usage for all future versions, and the issues copilot are raising can be properly addressed in the individual PRs on those branches.
